src/views/checkups/Checkups.tsx

```diff
@@ -1,23 +1,34 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { RouteComponentProps } from 'react-router-dom';
 
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import { ListPageHeader } from '@openshift-console/dynamic-plugin-sdk';
 import { Tab, Tabs, TabTitleText } from '@patternfly/react-core';
+import { createURL } from '@virtualmachines/details/tabs/overview/utils/utils';
 
 import CheckupsNetworkList from './network/list/CheckupsNetworkList';
+import CheckupsStorageList from './storage/list/CheckupsStorageList';
+import { trimLastHistoryPath } from './utils/utils';
 import CheckupsRunButton from './CheckupsRunButton';
 
 import './checkups.scss';
 
-const CheckupsList: FC<RouteComponentProps> = (props) => {
+const CheckupsList: FC<RouteComponentProps> = ({ history }) => {
   const { t } = useKubevirtTranslation();
-  const [activeTabKey, setActiveTabKey] = useState<number | string>(0);
+  const [activeTabKey, setActiveTabKey] = useState<number | string>(
+    history?.location?.pathname.endsWith('storage') ? 1 : 0,
+  );
+
+  useEffect(() => {
+    history.push(
+      createURL(activeTabKey === 0 ? 'network' : 'storage', trimLastHistoryPath(history)),
+    );
+  }, [activeTabKey, history, history?.location?.pathname]);
 
   return (
     <>
       <ListPageHeader title={t('Checkups')}>
-        <CheckupsRunButton {...props} />
+        <CheckupsRunButton history={history} />
```

Comment:
Seems like network and storage can be defined as parameters in the path.
So that you can grab them using `match.params` from the `RouteComponentProps` props 

Like  `/k8s/ns/:ns/checkups/:type/form`. Would we insert other types of Checkups in the future?

Reply:
no plans for more right now.
it wont really give me anything adding :type as params, as each one is a different component, and console-extention is managing the lins and component according to links, except Checkups.tsx which function a bit as a router because of the tabs which I don't have a way around it.